src/Symfony/Component/Notifier/Bridge/ClickSend/ClickSendTransport.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\ClickSend;
+
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author gnito-org <https://github.com/gnito-org>
+ */
+final class ClickSendTransport extends AbstractTransport
+{
+    protected const HOST = 'rest.clicksend.com';
+
+    public function __construct(
+        private readonly string $apiUsername,
+        #[\SensitiveParameter] private readonly string $apiKey,
+        private readonly ?string $from = null,
+        private readonly ?string $source = null,
+        private readonly ?string $listId = null,
+        private readonly ?string $fromEmail = null,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        $queryParameters = [];
+        if ($this->from) {
+            $queryParameters['from'] = $this->from;
+        }
+        if ($this->source) {
+            $queryParameters['source'] = $this->source;
+        }
+        if ($this->listId) {
+            $queryParameters['list_id'] = $this->listId;
+        }
+        if ($this->fromEmail) {
+            $queryParameters['from_email'] = $this->fromEmail;
+        }
+
+        return sprintf('clicksend://%s', $this->getEndpoint()).($queryParameters ? '?'.http_build_query($queryParameters) : null);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
```

Comment:
Please check for the corresponding Options class too, like in other bridges. Thanks

Reply:
Please point me to the one that has the best example, in your opinion.